### PR TITLE
Load model by full path correctly.

### PR DESCRIPTION
**Pull request by joejuzl**
**Proposed changes**:
We were always loading the latest model in a directory even if given the full path to the model archive. This fixes this so now we will load directly in the path is a file.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by tayfun — APPROVED** [review]

File: rasa/core/processor.py [hunk 1]
```diff
@@ -105,9 +105,12 @@ def _load_model(
...
         """Unpacks a model from a given path using the graph model loader."""
-        model_tar = get_latest_model(model_path)
-        if not model_tar:
-            raise ModelNotFound(f"No model found at path '{model_path}'.")
+        if os.path.isfile(model_path):
+            model_tar = model_path
+        else:
+            model_tar = get_latest_model(model_path)
+            if not model_tar:
+                raise ModelNotFound(f"No model found at path '{model_path}'.")
```
Comment (wochinge) [inline rasa/core/processor.py:113]: should this be a separate method? In my opinion we should avoid that we have a similar bug in other modules using `get_latest_model`

Comment (joejuzl) [reply rasa/core/processor.py:113]: I was going to change `get_latest_model` but looking at it, its pretty clear in its function - I think I just miss-used it here.

Comment (ka-bu) [reply rasa/core/processor.py]: Could we just raise in `get_latest_model` if what is passed is already a file? I don't find that behaviour of falling back to the dirname in that case is that obvious

Comment (joejuzl) [reply rasa/core/processor.py]: > I don't find that behaviour of falling back to the dirname in that case is that obvious

I also think the behaviour is weird. However I didn't want to affect other uses of this method in the codebase this close to the release. So I opted to just fixing how we load in the `Processor`.

**Comment by melindaloubser1** [comment]
I think this is also what caused the replace model endpoint to reload the latest model instead of the one at the provided path - sound right? https://github.com/RasaHQ/rasa/issues/10148#issuecomment-969350009

**Comment by joejuzl** [comment]
> I think this is also what caused the replace model endpoint to reload the latest model instead of the one at the provided path - sound right? #10148 (comment)

Yes that makes sense 👍

File: tests/core/test_processor.py [hunk 2]
```diff
@@ -1496,3 +1498,26 @@ async def test_model_name_is_available(trained_moodbot_path: Text):
     processor = Agent.load(model_path=trained_moodbot_path).processor
     assert len(processor.model_filename) > 0
     assert "/" not in processor.model_filename
+
+
+async def test_loads_correct_model_from_path(
+    trained_core_model: Text, trained_nlu_model: Text, tmp_path
```
Comment (wochinge) [inline tests/core/test_processor.py:1504]:
```suggestion
    trained_core_model: Text, trained_nlu_model: Text, tmp_path: Path
```

File: rasa/core/processor.py [hunk 3]
```diff
@@ -105,9 +105,15 @@ def _load_model(
         model_path: Union[Text, Path]
     ) -> Tuple[Text, ModelMetadata, GraphRunner]:
         """Unpacks a model from a given path using the graph model loader."""
-        model_tar = get_latest_model(model_path)
-        if not model_tar:
-            raise ModelNotFound(f"No model found at path '{model_path}'.")
+        try:
+            if os.path.isfile(model_path):
+                model_tar = model_path
```
Comment (wochinge) [inline rasa/core/processor.py:110]: Does this change require adding / updating a test?

**Review by wochinge — APPROVED** [review]

File: rasa/core/processor.py (hunk 3 above)
Comment (joejuzl) [reply rasa/core/processor.py:110]: the new test `test_loads_correct_model_from_path` breaks before this is added.

**Pull request merged**